### PR TITLE
Feat audio

### PR DESCRIPTION
**Pull request by apachano**
Resolves: #327 
Authors:
@apachano 

## Summary of changes
- Implement Soloud library
- Implement Background music
- Implement API for registering sounds
- Implement API for triggering sounds on block place event
- Implement API for triggering sounds on block break event
  
## Caveats
Does this introduce any new bugs?
- Until an OSX bug is merged into Soloud's master branch, we are sourcing from a forked version of the repo
- Audio isn't 3D yet

## On approval
Merge

### CODE REVIEW

**Comment by tobyplowy** [comment]
Mac Ci fails. Always mac for whatever reason.

**Comment by apachano** [comment]
@tobyplowy The submodule BS is fixed

**Comment by tobyplowy** [comment]
PR fails to build for me. 
[message.txt](https://github.com/GentenStudios/Phoenix/files/5894125/message.txt)

**Review by tobyplowy — APPROVED** [review]

File: Phoenix/Client/Include/Client/Voxels/AudioEventHandler.hpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+// POSSIBILITY OF SUCH DAMAGE.
+
+#pragma once
+
+#include <Client/AudioRegistry.hpp>
+#include <Client/Voxels/BlockRegistry.hpp>
+
+#include <Common/Voxels/Map.hpp>
+
+namespace phx::client
```
Comment (beeperdeeper089) [inline Phoenix/Client/Include/Client/Voxels/AudioEventHandler.hpp:36]: Doc this.

File: Phoenix/ThirdParty/CMakeLists.txt [hunk 2]
```diff
@@ -9,15 +9,22 @@ set(JSON_BuildTests OFF CACHE INTERNAL "")
...
 
-add_subdirectory (SDL2)
-add_subdirectory (Glad)
-add_subdirectory (ImGui)
-add_subdirectory (sol2)
-add_subdirectory (lua)
-add_subdirectory (enet)
-add_subdirectory (entt)
+# Set the SoLoud backend
+set(WITH_SDL2 ON)
```
Comment (beeperdeeper089) [inline Phoenix/ThirdParty/CMakeLists.txt:13]: We don't have to use SDL?

[This](https://sol.gfxile.net/soloud/basics.html) shows that we can probably get away with it using default system backends - if you can get it compiling (I remember spending a while getting that working when I tried this library out for the first time).

**Review by beeperdeeper089 — CHANGES_REQUESTED** [review]
Mainly documentation and we should be able to not need to use the SDL audio backend - since I think it's not very good anyways

File: Phoenix/Client/Include/Client/Voxels/BlockRegistry.hpp [hunk 3]
```diff
@@ -233,10 +242,58 @@ namespace phx::client
...
+					    for (const auto& sourceID : *luaSoundOnBreak)
+					    {
+						    auto* source = m_audioRegistry->getByID(sourceID);
+						    if (source != nullptr)
+						    {
+							    onBreakSources.push_back(source);
+						    }
+						    else
+						    {
+							    LOG_WARNING("MODDING")
```
[the remark this answers is not in the document]
Comment (apachano) [reply Phoenix/Client/Include/Client/Voxels/BlockRegistry.hpp:264]: This has been a complaint since day 1 for me, word wrap is a thing for most IDEs.

**Comment by apachano** [comment]
@tobyplowy Can you re-QA to make sure the automatic audiobackend search works. Volume is a touch low on Linux but good enough for now.

**Comment by tobyplowy** [comment]
How do I test the automatic audio back end search ? 

Also PR builds and runs.

**Comment by apachano** [comment]
All tested lol, I switched the backend in the source so if it runs it's
good.

**Comment by beeperdeeper089** [comment]
I'll run clang-tidy on this and approve once I've done it in the morning

**Review by beeperdeeper089 — APPROVED** [review]

**Comment by beeperdeeper089** [comment]
"morning" being completely metaphorical

**Comment by beeperdeeper089** [comment]
but, merge conflict so have fun

**Review by beeperdeeper089 — APPROVED** [review]

**Pull request merged**